### PR TITLE
[BUGS-7429] Add filter to exclude select post types from purge

### PR DESCRIPTION
**Pull request by jazzsequence**
Update the Purger class to add a filter that allows users skip purges if a given post type is being updated.

This PR has been updated to include various CI tweaks. 

* we're using [wpunit-helpers](https://github.com/pantheon-systems/wpunit-helpers)
* I've broken `validate-fixture-version.sh` into a separate [gist](https://gist.github.com/jazzsequence/13f2493785ee8760323deae9ca96d09a) with the goal of standardizing that script and making it more robust
* since the filter is a feature add, I've bumped the versions everywhere to `1.5.0-dev`

### CODE REVIEW

File: .github/workflows/lint-test.yml [hunk 1]
```diff
@@ -79,7 +79,12 @@ jobs:
           path: ~/vendor
           key: test-dependencies-{{ checksum "composer.json" }}
           restore-keys: test-dependencies-{{ checksum "composer.json" }}
-      - name: Install dependencies
+      - name: Install WP-CLI
+        run: |
+          curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar
+          chmod +x wp-cli.phar
+          sudo mv wp-cli.phar /usr/local/bin/wp
```
Comment (pwtyler) [inline .github/workflows/lint-test.yml]:
```suggestion
      - name: Setup WP-CLI
         uses: godaddy-wordpress/setup-wp-cli@1
```
godaddy's got you.

File: README.md [hunk 2]
```diff
@@ -320,7 +353,9 @@ Pantheon Advanced Page Cache integrates with WordPress plugins, including:
 See [CONTRIBUTING.md](https://github.com/pantheon-systems/pantheon-advanced-page-cache/blob/master/CONTRIBUTING.md) for information on contributing.
 
 ## Changelog ##
-### 1.4.3-dev ###
+### 1.4.3-dev ###
```
Comment (pwtyler) [inline README.md]:
```suggestion
### 1.5.0-dev ###
```
Is it a feature or a bug fix?

**Review by pwtyler — APPROVED** [review]
This is superfluous if we're going to ship a release immediately, but in general, prefer to use the `x.y.z-dev` version in documentation and docblocks, then search-replace as part of release prep. It could sit on develop and then subsequent feature bump to a minor/major vn, "@since" will then point to a non-existent version.

if we ever get to implementing, plugin release actions _should_ handle inside php files already.

File: README.md (hunk 2 above)
Comment (jazzsequence) [reply README.md]: good point. feature, probably.

**Pull request merged**